experiments/compositions/composition/api/v1/composition_types.go

```diff
@@ -32,10 +32,11 @@ type ResourceRef struct {
 	// resource: ServiceIdentity.serviceusage.cnrm.cloud.google.com/v1beta1//sqladmin.googleapis.com
 
 	// OPTION 2
-	Group   string `json:"group" protobuf:"bytes,1,name=group"`
-	Version string `json:"version,omitempty" protobuf:"bytes,2,opt,name=version"`
-	Kind    string `json:"kind" protobuf:"bytes,3,name=kind"`
-	Name    string `json:"name" protobuf:"bytes,4,name=name"`
+	Group      string `json:"group" protobuf:"bytes,1,name=group"`
```

Comment:
Do we need the proto annotations?

Reply:
may be not. I started off with @cheftako's initial code and continued having it. 
We can remove if not needed for k8s usecase.

Reply:
Added a TODO in the file.